adapters/coap-vertx-base/src/main/java/org/eclipse/hono/adapter/coap/CoapContext.java

```diff
@@ -158,24 +159,37 @@ public String getAuthId() {
     /**
      * Sets a timer to trigger the sending of a separate ACK to a device.
      * <p>
+     * This method first tries to find a {@value CoapConstants#TIMEOUT_TO_ACK} property value
+     * in the tenant's CoAP adapter configuration.
+     * If no such property is found, the value given in the timeoutMillis parameter is used instead.
      *
-     * @param vertx Vertx to schedule the timer
+     * @param vertx The vert.x instance to set the timer on.
+     * @param tenant The tenant that the device belongs to which the request being processed originates from.
      * @param timeoutMillis The number of milliseconds to wait for a separate ACK. {@code -1}, never use separate
      *            response, {@code 0}, always use separate response.
+     * @throws NullPointerException if vertx or tenant are {@code null}.
      */
     public void startAcceptTimer(
             final Vertx vertx,
+            final TenantObject tenant,
             final long timeoutMillis) {
 
+        Objects.requireNonNull(vertx, "vert.x");
+        Objects.requireNonNull("tenant");
+
+        final long ackTimeout = Optional.ofNullable(tenant.getAdapter(Constants.PROTOCOL_ADAPTER_TYPE_COAP))
+                .map(config -> (Long) config.getExtensions().get(CoapConstants.TIMEOUT_TO_ACK))
```

Comment:
Given a JSON string with the "timeoutToAck" value set to something <= `Integer.MAX_VALUE`, this will lead to a `ClassCastException` because the Map value will be an `Integer`, not a `Long`. Using the primitive `long` type for the cast would work.

Besides that, I think it would be better to deal with a possible configuration error here, the "timeoutToAck" value being defined as a `String` for example. For the standard TenantObject fields, we deal with that gracefully, catching ClassCastExceptions via `JsonHelper.getValue`. I think for extension values, we should do the same, so that there can't be any ClassCastException thrown here.